examples/tls-logger/server.cpp

```diff
@@ -42,11 +49,13 @@ class server
                 buffer.data(), 
                 static_cast<uint32_t>(buffer.size()));
 
+	    file_lock.lock();
             file.write(reinterpret_cast<const char*>(buffer.data()),
                 buffer.size());
             
             file << std::endl;
-        }
+	    file_lock.unlock();
```

Comment:
You might want to call `file.close()` here just for the sake of correctness but I don't think it is 100% necessary since you already flushed the buffer.